src/org/sosy_lab/common/configuration/ConfigurationBuilder.java

```diff
@@ -260,9 +262,26 @@ public ConfigurationBuilder loadFromResource(Class<?> contextClass, String resou
     @Var Path sourcePath;
     @Var String sourceString;
     try {
+      // try to access the requested file.
+      URI uri = url.toURI();
+      if ("jar".equals(uri.getScheme())) {
+        // if packed in a JAR file, we try to register the JAR as file system.
+        for (FileSystemProvider provider : FileSystemProvider.installedProviders()) {
+          if (provider.getScheme().equalsIgnoreCase("jar")) {
+            try {
+              provider.getFileSystem(uri);
```

Comment:
Calling `getFileSystem` and throwing away its result is confusing, as this method should not have a side effect. Is the only reason why we call this method to determine whether the file system already exists?

In this case, I would find the following code much clearer:
```java
// Ensure file system for uri exists
try {
  provider.newFileSystem(uri, ImmutableMap.of());
} catch (FileSystemNotFoundException ignored) {
  // already exists
}
```

Reply:
Opening a new filesystem registers a FileSystemProvider, thus the URI can be resolved.
This seems to be some background magic of Java.

I agree with your code change and will change it.